watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2_P_extension.py

```diff
@@ -0,0 +1,627 @@
+
+import pyomo.environ as pyo
+from pyomo.network import Arc, SequentialDecomposition
+
+from idaes.core import (
+    FlowsheetBlock,
+)
+from idaes.models.unit_models import (
+    CSTR,
+    Feed,
+    Separator,
+    Product,
+    Mixer,
+    PressureChanger,
+)
+from idaes.models.unit_models.separator import SplittingType
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+import idaes.core.util.scaling as iscale
+from idaes.core.util.tables import (
+    create_stream_table_dataframe,
+    stream_table_dataframe_to_string,
+)
+from watertap.unit_models.cstr_injection import CSTR_Injection
+from watertap.property_models.anaerobic_digestion.modified_adm1_properties import (
+    ModifiedADM1ParameterBlock,
+)
+from watertap.property_models.anaerobic_digestion.adm1_properties_vapor import (
+    ADM1_vaporParameterBlock,
+)
+from watertap.property_models.anaerobic_digestion.modified_adm1_reactions import (
+    ModifiedADM1ReactionParameterBlock,
+)
+from watertap.property_models.activated_sludge.modified_asm2d_properties import (
+    ModifiedASM2dParameterBlock,
+)
+from watertap.property_models.activated_sludge.modified_asm2d_reactions import (
+    ModifiedASM2dReactionParameterBlock,
+)
+from watertap.unit_models.translators.translator_adm1_asm2d import (
+    Translator_ADM1_ASM2D,
+)
+from idaes.models.unit_models.mixer import MomentumMixingType
+from watertap.unit_models.translators.translator_asm2d_adm1 import Translator_ASM2d_ADM1
+from watertap.unit_models.anaerobic_digestor import AD
+from watertap.unit_models.dewatering import (
+    DewateringUnit,
+    ActivatedSludgeModelType as dewater_type,
+)
+from watertap.unit_models.thickener import (
+    Thickener,
+    ActivatedSludgeModelType as thickener_type,
+)
+from watertap.core.util.initialization import check_solve
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+def main():
+    m = build_flowsheet()
+    set_operating_conditions(m)
+    for mx in m.fs.mixers:
+        mx.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 3].deactivate()
+    print(f"DOF before initialization: {degrees_of_freedom(m)}")
+
+    initialize_system(m)
+    for mx in m.fs.mixers:
+        mx.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 3].deactivate()
+    print(f"DOF before initialization: {degrees_of_freedom(m)}")
+
+    results = solve(m)
+
+    # Switch to fixed KLa in R3 and R4 (S_O concentration is controlled in R5)
+    m.fs.R5.KLa.fix(240)
+    m.fs.R6.KLa.fix(240)
+    m.fs.R7.KLa.fix(84)
+    m.fs.R5.outlet.conc_mass_comp[:, "S_O2"].unfix()
+    m.fs.R6.outlet.conc_mass_comp[:, "S_O2"].unfix()
+    m.fs.R7.outlet.conc_mass_comp[:, "S_O2"].unfix()
```

Comment:
```suggestion
    # Switch to fixed KLa in R5, R6, and R7 (S_O concentration is controlled in R5)
    m.fs.R5.KLa.fix(240)
    m.fs.R6.KLa.fix(240)
    m.fs.R7.KLa.fix(84)
    m.fs.R5.outlet.conc_mass_comp[:, "S_O2"].unfix()
    m.fs.R6.outlet.conc_mass_comp[:, "S_O2"].unfix()
    m.fs.R7.outlet.conc_mass_comp[:, "S_O2"].unfix()
```
Am I misinterpreting this?